python/ray/data/dataset.py

```diff
@@ -2628,6 +2628,75 @@ def write_tfrecords(
             block_path_provider=block_path_provider,
         )
 
+    @ConsumptionAPI
```

Comment:
Shall we also annotate this also as `@PublicAPI(stability="alpha")` for a couple releases?

Reply:
Do you want me to make that change?

Reply:
Sounds good.

Reply:
Done.